src/App.tsx

```diff
@@ -26,70 +27,134 @@ type ReorderOptions = {
+  switch (sortType) {
+    case SortType.LENGTH:
+      visibleGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    case SortType.ALPHABET:
+      visibleGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export class App extends Component<{}, State> {
+  state = {
+    sortType: SortType.NONE,
+    isReversed: false,
+  };
+
+  sortAlphabet = () => this.setState(() => ({
+    sortType: SortType.ALPHABET,
+  }));
+
+  sortByLength = () => this.setState(() => ({
+    sortType: SortType.LENGTH,
+  }));
+
+  getReverse = () => this.setState((state: State) => ({
+    isReversed: !state.isReversed,
+  }));
+
+  getReset = () => this.setState(() => ({
+    sortType: SortType.NONE,
+    isReversed: false,
+  }));
+
+  render(): ReactNode {
+    const { isReversed, sortType } = this.state;
+    const {
+      sortAlphabet,
+      sortByLength,
+      getReverse,
+      getReset,
+    } = this;
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames(
+              'button',
+              'is-info',
+              {
+                'is-light': sortType !== SortType.ALPHABET,
+              },
+            )}
+            onClick={sortAlphabet}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames(
+              'button',
+              'is-success',
+              {
+                'is-light':
+                  sortType !== SortType.LENGTH,
+              },
+            )}
+            onClick={sortByLength}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classNames(
+              'button',
+              'is-warning',
+              {
+                'is-light': !isReversed,
+              },
```

Comment:
Write this way if line length allows.
```suggestion
              { 'is-light': !isReversed, },
```